packages/flutter/lib/src/material/switch.dart

```diff
@@ -1412,20 +1434,32 @@ class _SwitchPainter extends ToggleablePainter {
       return TweenSequence<Size>(thumbSizeSequence).animate(positionController);
     }
 
-    Size thumbSize;
-    if (reaction.isCompleted) {
-      thumbSize = Size.fromRadius(pressedThumbRadius);
+    Size? thumbSize;
+    if (isCupertino) {
+      if (reaction.isCompleted) {
+        thumbSize = Size(_pressedInactiveThumbRadius! * 2 + _pressedThumbExtention!, _pressedInactiveThumbRadius! * 2);
+      } else {
+        if (position.isDismissed || position.status == AnimationStatus.forward) {
+          thumbSize = Size.lerp(inactiveThumbSize, activeThumbSize, position.value);
+        } else {
+          thumbSize = Size.lerp(inactiveThumbSize, activeThumbSize, position.value);
+        }
+      }
     } else {
-      if (position.isDismissed || position.status == AnimationStatus.forward) {
-        thumbSize = thumbSizeAnimation(true).value;
+      if (reaction.isCompleted) {
+        thumbSize = Size.fromRadius(pressedThumbRadius);
       } else {
-        thumbSize = thumbSizeAnimation(false).value;
+        if (position.isDismissed || position.status == AnimationStatus.forward) {
+          thumbSize = thumbSizeAnimation(true).value;
+        } else {
+          thumbSize = thumbSizeAnimation(false).value;
+        }
       }
     }
 
     // The thumb contracts slightly during the animation in Material 2.
```

Comment:
Is this comment still valid? I don't see any `useMaterial3` checks here.

Reply:
Ah yes, the computation below `1.0 - (currentValue - thumbOffset!).abs() * 2.0` is for M2. The comment here is to explan why we have the computation:) M3 thumbOffset is null.